Datagrid: Console error when running clearFilter

**Describe the bug**
One of our testers often get an issue with a console log _"[ERROR] Exception in core.raise TypeError: Unable to get property 'clearFilter' of undefined or null reference" or "ERROR TypeError: Cannot read property 'clearFilter' of null"._ that shows up in Homepages Administration. I cannot reproduce it, but I have investigated it and found that it comes from the clearFilter method in soho-datagrid.component.ts where this.datagrid is undefined: 
```
  /**
   * Clears any filter defined, for this datagrid.
   */
  clearFilter(): void {
    this.ngZone.runOutsideAngular(() => {
      this.datagrid.clearFilter();
    });
  }
```

So my question is if you can please add a check to see if this.datagrid is null before calling clearFilter? We already have this test in Homepages before calling this method. Perhaps it's a timing issue so that it's not null when we call the ClearFilter method, but sometimes its null when runOutsideAngular is executed.

**To Reproduce**
Not easily reproducible :(

**Expected behavior**
It would be good if the method had a null check so it doesn't throw an error if this.datagrid is null/undefined when the clearfilter method should be executed
```
	clearFilter(): void {
		this.ngZone.runOutsideAngular(() => {
			if (this.datagrid) {
				this.datagrid.clearFilter();
			}
		});
	}
```

**Version**
- ids-enterprise-ng: 5.0

**Screenshots**
If applicable, add screenshots to help explain your problem.
![image](https://user-images.githubusercontent.com/8664975/53331648-b5360180-38f1-11e9-83ce-5b9c3ffbaa17.png)


**Platform**
 - OS Version: Windows 10
 - Browser Name: chrome / edge
 - Browser Version: Version 72.0.3626.109 (Official Build) (64-bit)

**Additional context**
Add any other context about the problem here.

Thanks!

--- a/docs/CHANGELOG.md
+++ b/docs/CHANGELOG.md
@@ -9,6 +9,7 @@
 - `[DataGrid]` 4.0 Datagrid expose error row clear functions `PWP` ([#314](https://github.com/infor-design/enterprise-ng/issues/314))
 - `[DataGrid]` Added a few missing toolbar options to the types. `TJM` ([#336](https://github.com/infor-design/enterprise-ng/issues/336))
 - `[DataGrid]` Added headerAlign option to datagrid column. `PWP` ([#304](https://github.com/infor-design/enterprise-ng/issues/304))
+- `[DataGrid]` Added safety check on clearFilter for random errors. `TJM` ([#374](https://github.com/infor-design/enterprise-ng/issues/374))
 - `[DatePicker]` Added firstDayOfWeek option to date picker component `PWP` ([362](https://github.com/infor-design/enterprise-ng/issues/362)'
 - `[General]` Upgraded @angular/cli (to 7.3.2) and @angular/core (to 7.2.5).  `BTHH` ([Pull Request 386](https://github.com/infor-design/enterprise-ng/pull/386))
 

--- a/projects/ids-enterprise-ng/src/lib/datagrid/soho-datagrid.component.ts
+++ b/projects/ids-enterprise-ng/src/lib/datagrid/soho-datagrid.component.ts
@@ -1419,7 +1419,9 @@ export class SohoDataGridComponent implements OnInit, AfterViewInit, OnDestroy, 
    */
   clearFilter(): void {
     this.ngZone.runOutsideAngular(() => {
-      this.datagrid.clearFilter();
+      if (this.datagrid) {
+        this.datagrid.clearFilter();
+      }
     });
   }
 

